framework/src/distributions/BasicUniformDistribution.C

```diff
@@ -0,0 +1,67 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "BasicUniformDistribution.h"
+
+template<>
+InputParameters validParams<BasicUniformDistribution>()
```

Comment:
I think just rename this to UniformDistribution. I know we talked about maybe needing a "basic" and "advanced" distributions, but based on what I am seeing I don't think we will. Like I mentioned in conversation, if people need to create a distribution and don't need to the cdf/pdf/inverseCdf funtion they can just return 0 for those cases.

We might want the base class to produce errors for those functions rather than be pure virtual, but I am still not sure on that, so just leave them as pure virtual.